OpenRA.Mods.Common/Installer/SourceResolvers/SteamSourceResolver.cs

```diff
@@ -0,0 +1,148 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Microsoft.Win32;
+
+namespace OpenRA.Mods.Common.Installer
+{
+	public class SteamSourceResolver : ISourceResolver
```

Comment:
Is it too much scope creep to ask you to add this to TD and RA to copy the legacy assets out of the remaster's `Data/CNCDATA/RED_ALERT/CD1` (note: only has the allies disk, so we can't install soviet FMVs) and `Data/CNCDATA/TIBERIAN_DAWN/CD1`, `Data/CNCDATA/TIBERIAN_DAWN/CD2` respectively?

Reply:
No problem! Ill add that ;)